cpp/include/raft/neighbors/sample_filter.cuh

```diff
@@ -45,4 +45,34 @@ struct bitset_filter {
     return bitset_view_.test(sample_ix);
   }
 };
+
+/**
+ * @brief Filter used to convert the cluster index and sample index
+ * of an IVF search into a sample index. This can be used as an
+ * intermediate filter.
+ *
+ * @tparam index_t Indexing type
+ * @tparam filter_t
+ */
+template <typename index_t, typename filter_t>
+struct ivf_to_sample_filter {
```

Comment:
In order to provide a unified filtering interface, would it make sense to wrap any filter that is passed to `search_with_filtering` for IVF methods in this struct? I have trouble imagining a scenario in which the outer search would really want to make use of a filter that takes into account the in-cluster ID rather than the overall ID, and that would establish a rule  that the filter's signature is just `(query_index, sample_index) -> bool`.

Reply:
@alexanderguzhva is the original creator of this filtering for IVF indexes. Do you have any examples or use-cases of filtering applied to a cluster/ids in the cluster?
If we can't find a use-case then I think it make sense to wrap every filter with `ivf_to_sample_filter` for a consistent filtering API.